cc2538-bsl.py

```diff
@@ -784,11 +784,12 @@ def read(filename):
                 length = length + (length % 4)
```

Comment:
`length = (length + 4) - (length % 4)` Isn't correct either, it adds 4 even if its already a multiple. I need sleep.

Yours look pretty fancy indeed :D
This one seems to work too `x -= x % -4`, your pick.
The whole `if` statement can go too then.